QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+
+qspi_status_t QSPIFBlockDevice::_qsp_set_frequency(int freq)
+{
+    return _qspi.set_frequency(freq);
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_send_read_command(unsigned int read_inst, void *buffer, bd_addr_t addr,
+        bd_size_t size)
+{
+    // Send Read command to device driver
+    size_t bufLen = size;
+
+    if (_qspi.read(read_inst, -1, (unsigned int )addr, (char *)buffer, &bufLen) != QSPI_STATUS_OK ) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Read failed");
+        return QSPI_STATUS_ERROR;
+    }
+
+    return QSPI_STATUS_OK;
+
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_send_program_command(unsigned int progInst, const void *buffer, bd_addr_t addr,
+        bd_size_t *size)
+{
+    // Send Program (write) command to device driver
+    qspi_status_t result = QSPI_STATUS_OK;
+
+    result = _qspi.write(progInst, -1, addr, (char *)buffer, (size_t *)size);
+    if (result != QSPI_STATUS_OK) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: QSPI Write failed");
+    }
+
+    return result;
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_send_erase_command(unsigned int eraseInst, bd_addr_t addr, bd_size_t size)
+{
+    // Send Erase Instruction command to driver
+    qspi_status_t result = QSPI_STATUS_OK;
+
+    QSPIF_LOG(QSPIF_DEBUG_INFO, "INFO: Inst: 0x%xh, addr: %llu, size: %llu", eraseInst, addr, size);
+
+    result = _qspi.command_transfer(eraseInst, // command to send
+                                    (((int)addr) & 0x00FFF000), // Align addr to 4096
+                                    NULL,                 // do not transmit
+                                    0,              // do not transmit
+                                    NULL,                 // just receive two bytes of data
+                                    0); // store received values in status_value
+
+    if (QSPI_STATUS_OK != result) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: QSPI Erase failed");
+    }
+
+    return result;
+
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_send_general_command(unsigned int instruction, bd_addr_t addr,
+        const char *tx_buffer,
+        size_t tx_length, const char *rx_buffer, size_t rx_length)
+{
+    // Send a general command Instruction to driver
+    qspi_status_t status = _qspi.command_transfer(instruction, (int)addr, tx_buffer, tx_length, rx_buffer, rx_length);
+
+    if (QSPI_STATUS_OK != status) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR:Sending Generic command: %x", instruction);
+    }
+
+    return status;
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_configure_format(qspi_bus_width_t inst_width, qspi_bus_width_t address_width,
+        qspi_address_size_t address_size, qspi_bus_width_t alt_width, qspi_alt_size_t alt_size, qspi_bus_width_t data_width,
+        int dummy_cycles)
+{
+    // Configure QSPI driver Bus format
+    _qspi.configure_format(inst_width, address_width, address_size, alt_width, alt_size, data_width, dummy_cycles);
```

Comment:
Why is the return value ignored and always returns OK (line 1242) ?

Reply:
I saw that driver's 'configure_format' always return OK so I thought not to take its returned value..  Its my mistake I'll take it, just in case the driver changes one day.